src/desktop/apps/authentication/helpers.ts

```diff
@@ -74,8 +75,14 @@ export const handleSubmit = (
         analytics.track(action, pickBy(properties, identity))
       }
 
-      const defaultRedirect = getRedirect(type)
-      window.location = modalOptions.redirectTo || (defaultRedirect as any)
+      let afterAuthURL: URL
+      if (modalOptions.redirectTo)
+        afterAuthURL = new URL(modalOptions.redirectTo, sd.APP_URL)
+      else afterAuthURL = getRedirect(type)
+
+      const result = await apiAuthWithRedirectUrl(res, afterAuthURL)
```

Comment:
Minor comment on naming. If I understand correctly, `apiAuthWithRedirectUrl` doesn't actually sign in to Gravity. It happens when following the redirect below. I like that this method always returns a `URL`, so maybe this can be something like

```javascript
const url = await redirectUrlAfterAuth(res, afterAuthURL)
window.location.href = url.href
```

and add a comment on the method mentioning it will try to return a url for signing in to Gravity.

Reply:
Hmm - so while `redirectUrlAfterAuth(res, afterAuthURL)` returns a `URL` object – visiting the href of that object _is the request that authenticates the user to Gravity_ – so I believe that this is correct. Happy to chat in real-time about this, just lmk.

Reply:
Yeah I meant signing in to Gravity doesn't happen in calling `apiAuthWithRedirectUrl()` but in `window.location.href = result.href` below. When I first read the method name, it wasn't immediately clear.

I don't feel too strongly about this comment and I couldn't come up with a good name. What the method does is something like "return a URL that can authenticate the user to api and redirect back to the app". Feel free to move forward with your call!